### DISCUSSION TITLE
status: MODEL_INVALID on 9.1

### DISCUSSION THREAD

**Discussion by jweizman** [google/or-tools#2810] [CP-SAT questions] [answered]
Hi

My model worked perfectly until 9.0 and broke on 9.1.

Starting CP-SAT solver.
Parameters: max_time_in_seconds: 60 log_search_progress: true num_search_workers: 16
Invalid model: The divisor cannot span across zero in constraint: int_div { target: 264 vars: 263 vars: 262 }

CpSolverResponse summary:
status: MODEL_INVALID

what shoud i send to help trouble shoot?
hx
Jonathan

**Comment by Mizux** [comment, marked as answer]
In v9.1 the model validator is more pedantic contrary to v9.0 (which was leading to undefined behaviour)
You should try to remove or fix your model to get rid of this divisor constraint issue (i.e. unsupported constraint).

**Comment by jweizman** [comment]
are you talking about model.AddDivisionEquality ?

it is not supported ? I'm not 100% sure i understant what should i do since the model IS correct?

**Reply by lperron** [reply]
the divisor cannot span across 0. Is it the case ?

**Reply by lperron** [reply]
If not, can you send me the model ?

**Reply by jweizman** [reply]
what does it means? (i'm french), what is span accross 0? In what form sould i send the model? C# code?

**Comment by lperron** [comment]
The min is <= 0 and the max if >= 0.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by jmarca** [reply]
In this case, I think the divisor must be strictly greater than zero

(I wrote the now-buggy code, the divisor is from 0 to N, needs to be 1 to N)

**Comment by jweizman** [comment]
Ah ok then we found.
Interesting that version 9 .0 never complained
Thx!

Le lun. 4 oct. 2021 à 17:31, Laurent Perron ***@***.***> a
écrit :

> The min is <= 0 and the max if >= 0.
> Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
> 00
>
>
>

**Comment by lperron** [comment]
As I said, we try to make the solver more pedantic, instead of silently
producing a wrong solution.

**Comment by jweizman** [comment]
I understand.
Thx

Le lun. 4 oct. 2021 à 21:45, Laurent Perron ***@***.***> a
écrit :

> As I said, we try to make the solver more pedantic, instead of silently
> producing a wrong solution.
>
